### PR TITLE
facets: experiments validation

### PR DESCRIPTION
**Pull request by Panos512**
* Trying to fix misspelled experiment names.

* Adds a new field 'facet_experiment' that should be used
  to generate facets.

Signed-off-by: Panos Paparrigopoulos <panos.paparrigopoulos@cern.ch>

### CODE REVIEW

**Comment by kaplun** [comment]
cc @jmartinm

File: inspire/modules/records/receivers.py [hunk 1]
```diff
@@ -1,30 +1,57 @@
...
+from invenio_records.signals import before_record_index, before_record_insert
+
+
+@before_record_index.connect
+def match_valid_experiments(recid, json):
+    """Matches misspelled experiment names with valid experiments.
+       Tries to match with valid experiments by matching lowercased and
+       whitespace-free versions of known experiments.
+    """
+    experiments_list_original = [
```
Comment (kaplun) [inline inspire/modules/records/receivers.py]: Can you add a FIXME comment here to remember ourselves this is just a quick hack and we should use a better method based on checking against the whole experiment list? (You could actually add a github issue on inspire-next related to that and reference it in the FIXME).

File: inspire/modules/records/receivers.py [hunk 2]
```diff
@@ -35,3 +32,33 @@ def populate_inspire_subjects(recid, json):
...
+    """Matches misspelled experiment names with valid experiments.
+       Tries to match with valid experiments by matching lowercased and
+       whitespace-free versions of known experiments.
+    """
+    # FIXME: This list is temporary. We should have a list that is generated
+    # from the current state of our data.
+    from .experiment_list import EXPERIMENTS_LIST
+    experiments_list_original = EXPERIMENTS_LIST
+    experiments_list = map(str.lower, experiments_list_original)
+    experiments_list = [exp.replace(' ', '') for exp in experiments_list]
```
Comment (kaplun) [inline inspire/modules/records/receivers.py]: Move this into `.experiment_list`.

File: inspire/modules/records/receivers.py [hunk 3]
```diff
@@ -35,3 +32,33 @@ def populate_inspire_subjects(recid, json):
...
+
+    experiments = json.get("accelerator_experiments")
+    if experiments:
```
Comment (kaplun) [inline inspire/modules/records/receivers.py]: You can actually move the import here, so that there's no computation in case of no experiment.

**Comment by kaplun** [comment]
@Panos512 ouch! Can you rebase once more this branch? Then I think it's good for :shipit:

**Comment by Panos512** [comment]
@kaplun ready!

**Pull request merged**